discovery-handlers/opcua/src/discovery_impl.rs

```diff
@@ -180,6 +183,32 @@ fn get_socket_addr(url: &str) -> Result<SocketAddr, anyhow::Error> {
     Ok(addr)
 }
 
+// This checks if the discovery_url can be resolved, if not use ip address instead
+fn get_discovery_url_ip(ip_url: &str, discovery_url: String) -> String {
+    let url = Url::parse(&discovery_url).unwrap();
+    let mut path = url.path().to_string();
+    let host = url.host_str().unwrap();
```

Comment:
I'm not really fond of panicking if an external service sends crappy things, I think it is better to directly return an `Option<String>` here and return `None` if the URL we got is completely invalid (and log an error so the user can be aware of that).